File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/screens/pages/articlesinfo/ArticlesInfoViewModel.java
package ru.usedesk.knowledgebase_gui.screens.pages.articlesinfo;

import androidx.annotation.NonNull;

import java.util.List;

import ru.usedesk.knowledgebase_gui.screens.common.DataViewModel;
import ru.usedesk.knowledgebase_gui.screens.common.ViewModelFactory;
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBase;
import ru.usedesk.knowledgebase_sdk.external.entity.UsedeskArticleInfo;

class ArticlesInfoViewModel extends DataViewModel<List<UsedeskArticleInfo>> {

    private ArticlesInfoViewModel(@NonNull IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk, long categoryId) {
        loadData(usedeskKnowledgeBaseSdk.getArticlesRx(categoryId));
    }

    static class Factory extends ViewModelFactory<ArticlesInfoViewModel> {
        private final IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk;
        private final long categoryId;

        public Factory(@NonNull IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk, long categoryId) {
            this.usedeskKnowledgeBaseSdk = usedeskKnowledgeBaseSdk;
            this.categoryId = categoryId;
        }

        @NonNull
        @Override
        protected ArticlesInfoViewModel create() {
            return new ArticlesInfoViewModel(usedeskKnowledgeBaseSdk, categoryId);
        }

        @NonNull
        @Override
        protected Class<ArticlesInfoViewModel> getClassType() {
            return ArticlesInfoViewModel.class;
        }
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/internal/screens/pages/articlesinfo/ArticlesInfoViewModel.kt
package ru.usedesk.knowledgebase_gui.internal.screens.pages.articlesinfo

import ru.usedesk.knowledgebase_gui.internal.screens.common.DataViewModel
import ru.usedesk.knowledgebase_gui.internal.screens.common.ViewModelFactory
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBase
import ru.usedesk.knowledgebase_sdk.external.entity.UsedeskArticleInfo

class ArticlesInfoViewModel private constructor(usedeskKnowledgeBaseSdk: IUsedeskKnowledgeBase, categoryId: Long) : DataViewModel<List<UsedeskArticleInfo?>?>() {
    internal class Factory(private val usedeskKnowledgeBaseSdk: IUsedeskKnowledgeBase, private val categoryId: Long) : ViewModelFactory<ArticlesInfoViewModel?>() {
        override fun create(): ArticlesInfoViewModel {
            return ArticlesInfoViewModel(usedeskKnowledgeBaseSdk, categoryId)
        }

        override fun getClassType(): Class<ArticlesInfoViewModel?> {
            return ArticlesInfoViewModel::class.java
        }
    }

    init {
        loadData(usedeskKnowledgeBaseSdk.getArticlesRx(categoryId))
    }
}